File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        store = {}
        max = 0
        store[0] = 0
        for num in nums:
            if num in store:
                store[num] += 1
                if (store[num] > store[max]):
                    max = num
                if store[num] > (len(nums) / 2):
                    return num
            else:
                store[num] = 1
                if (store[num] > store[max]):
                    max = num
        return max

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        if (s == "") {
            return 0;
        }
        int start = 0;
        int end = 0;
        int longestLength = 1;
        Dictionary<char, int> store = new Dictionary<char, int>();
        
        store[s[start]] = 1;

        while (end < s.Length) {
            if (end+1 >= s.Length) {
                break;
            }
            end++;
            if (!store.ContainsKey(s[end])) {
                store[(s[end])] = 1;
            }
            else {
                store[s[end]]++;
            }
            while (store[s[end]] > 1) {
                    store[s[start]]--;
                    start++;
                }
                longestLength = Math.Max(longestLength, end - start + 1);
        }
        return longestLength;

    }
}
